streams/shared/src/main/scala/zio/stream/ZStream.scala

```diff
@@ -2969,6 +2969,50 @@ abstract class ZStream[-R, +E, +O](val process: ZManaged[R, Nothing, ZIO[R, Opti
       } yield pull
     }
   }
+
+  /**
+  * Zips each element with the next element if present.
+  */
+  final def zipWithNext: ZStream[R, E, (O, Option[O])] =
+    ZStream {
+      for {
+        chunks <- self.process
+        ref    <- Ref.make[Option[O]](None).toManaged_
+        done   <- Ref.make(false).toManaged_
```

Comment:
The stream always terminates at the same moment as the original stream. 
Do we need `done` here?

Reply:
@simpadjo yes and no. Your question made me really I can just set `ref` to `None` instead of having an extra reference for termination. OTOH, without `done` or `*> ref.set(None)` it would get stuck in an infinite loop. Updating in a few minutes.

Reply:
@simpadjo done